src/main/java/com/binance/api/client/impl/BinanceApiWebSocketClientImpl.java

```diff
@@ -25,34 +25,40 @@ public BinanceApiWebSocketClientImpl() {
     this.client = new OkHttpClient();
   }
 
-  public void onDepthEvent(String symbol, BinanceApiCallback<DepthEvent> callback) {
+  public Closeable onDepthEvent(String symbol, BinanceApiCallback<DepthEvent> callback) {
     final String channel = String.format("%s@depth", symbol);
-    createNewWebSocket(channel, new BinanceApiWebSocketListener<>(callback, DepthEvent.class));
+    return createNewWebSocket(channel, new BinanceApiWebSocketListener<>(callback, DepthEvent.class));
   }
 
   @Override
-  public void onCandlestickEvent(String symbol, CandlestickInterval interval, BinanceApiCallback<CandlestickEvent> callback) {
+  public Closeable onCandlestickEvent(String symbol, CandlestickInterval interval, BinanceApiCallback<CandlestickEvent> callback) {
     final String channel = String.format("%s@kline_%s", symbol, interval.getIntervalId());
-    createNewWebSocket(channel, new BinanceApiWebSocketListener<>(callback, CandlestickEvent.class));
+    return createNewWebSocket(channel, new BinanceApiWebSocketListener<>(callback, CandlestickEvent.class));
   }
 
-  public void onAggTradeEvent(String symbol, BinanceApiCallback<AggTradeEvent> callback) {
+  public Closeable onAggTradeEvent(String symbol, BinanceApiCallback<AggTradeEvent> callback) {
     final String channel = String.format("%s@aggTrade", symbol);
-    createNewWebSocket(channel, new BinanceApiWebSocketListener<>(callback, AggTradeEvent.class));
+    return createNewWebSocket(channel, new BinanceApiWebSocketListener<>(callback, AggTradeEvent.class));
   }
 
-  public void onUserDataUpdateEvent(String listenKey, BinanceApiCallback<UserDataUpdateEvent> callback) {
-    createNewWebSocket(listenKey, new BinanceApiWebSocketListener<>(callback, UserDataUpdateEvent.class));
+  public Closeable onUserDataUpdateEvent(String listenKey, BinanceApiCallback<UserDataUpdateEvent> callback) {
+    return createNewWebSocket(listenKey, new BinanceApiWebSocketListener<>(callback, UserDataUpdateEvent.class));
   }
 
-  private void createNewWebSocket(String channel, BinanceApiWebSocketListener<?> listener) {
+  private Closeable createNewWebSocket(String channel, BinanceApiWebSocketListener<?> listener) {
     String streamingUrl = String.format("%s/%s", BinanceApiConstants.WS_API_BASE_URL, channel);
     Request request = new Request.Builder().url(streamingUrl).build();
-    client.newWebSocket(request, listener);
+    final WebSocket webSocket = client.newWebSocket(request, listener);
+    return () -> {
+      final int code = 1000;
```

Comment:
Any particular reason for code value 1000 ?

Reply:
`1000` is the status code for a normal ws shutdown, basically telling the server 'Thanks, I'm done'.